src/util/categoryUtils.js

```diff
@@ -0,0 +1,141 @@
+
+// Extract loan channel ids and loans query from featuredLoanHeroQuery result data
+function parseFeaturedLoanData(data) {
+	// use setting based id if available
+	const featuredChannelIdSetting = data?.general?.uiConfigSetting?.value ?? '';
+	const loanChannelIds = featuredChannelIdSetting
+		? [parseInt(featuredChannelIdSetting, 10)] : fallbackCategoryIds;
+	return { loanChannelIds, loansQuery: featuredLoansQuery };
+}
+
+/**
+ * Get the current Featured Loan Channel setting asynchronously.
+ *
+ * @param  {ApolloClient} client     the client to make the request to
+ * @return {Object}                  { loanChannelIds, loansQuery }
+ */
+export function getFeaturedLoanChannelAsync(client) {
+	// get featured loan channel from ui setting
+	return client.query({
+		query: featuredLoanHeroQuery
+	}).then(({ data }) => {
+		return parseFeaturedLoanData(data);
+	});
+}
+
+/**
+ * Get the current Featured Loan Channel setting synchronously.
+ *
+ * @param  {ApolloClient} client     the client to read the data from
+ * @return {Object}                  { loanChannelIds, loansQuery }
+ */
+export function getFeaturedLoanChannelCached(client) {
+	try {
+		// get setting driven featured loan channel
+		const data = client.readQuery({
+			query: featuredLoanHeroQuery,
+		});
+		return parseFeaturedLoanData(data);
+	} catch (e) {
+		logReadQueryError(e, 'FeatureHeroLoanWrapper featureLoanChannelSetting');
+		return {};
+	}
+}
+
+// Query for getting the current hero channel id from ML
+export const heroChannelIdQuery = gql`query heroChannelId {
+	ml {
+		heroChannel(platform: marketplace) {
+			id
+		}
+	}
+}`;
+
+// Extract loan channel ids and loans query from heroChannelIdQuery result data
+function parseHeroChannelData(data) {
+	// use ml heroChannel id if available
+	const { id, __typename } = data?.ml?.heroChannel ?? {};
+	const heroChannelId = parseInt(id, 10) || 0;
+	const loanChannelIds = heroChannelId ? [heroChannelId] : fallbackCategoryIds;
+	const loansQuery = __typename === 'RecLoanChannel' ? recommendedLoansQuery : featuredLoansQuery;
+	return { loanChannelIds, loansQuery };
+}
+
+/**
+ * Get the current Hero Channel id asynchronously.
+ *
+ * @param  {ApolloClient} client     the client to make the request to
+ * @return {Object}                  { loanChannelIds, loansQuery }
+ */
+export function getHeroChannelAsync(client) {
+	// get experiment assignment and then hero channel id query
+	return client.query({
+		query: experimentAssignmentQuery,
+		variables: { id: 'EXP-VUE-917-hero-loan' },
+	}).then(() => client.query({
+		query: heroChannelIdQuery
+	})).then(({ data }) => {
+		return parseHeroChannelData(data);
+	});
+}
+
+/**
+ * Get the current Hero Channel id synchronously.
+ *
+ * @param  {ApolloClient} client     the client to read the data from
+ * @return {Object}                  { loanChannelIds, loansQuery }
+ */
+export function getHeroChannelCached(client) {
+	try {
+		// get hero channel id from chache if available
```

Comment:
What are they keys in this cache, and what's its lifetime?  Are there docs for UI caching anywhere that I could read?

Reply:
Gotcha, thanks!